task5_Ruben_Chiche/src/main.cpp

```diff
@@ -0,0 +1,148 @@
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <List>
+#include <string>
+#include <vector>
+#include <algorithm>
+#include <sstream>
+#include <iterator>
+#include "Coordinate.h"
+#include "Distance.h"
+
+std::string PATH = "data.txt";
+
+void readData(std::string, std::map<std::string, Coordinate>&);
+
+int main() { 
+
+    std::map<std::string, Coordinate> cityToLocation;
+
+    // try reading the data from the data.txt file
+    try
+    {
+        readData(PATH, cityToLocation);
+    }
+    catch (const std::exception& e)
+    {
+        std::cout << e.what();
+        return 1;
+    }
+    
+    // mapping a number to a distance function
+    std::map<size_t, double (*)(Coordinate, Coordinate)> choiceToFunc;
+    choiceToFunc[0] = Distance::euclidean;
+    choiceToFunc[1] = Distance::chebyshev;
+    choiceToFunc[2] = Distance::manhattan;
+
+    double radius;
+    size_t choice;
+    Coordinate targetCoordinate;
+    std::string city;
+    
+    do
+    {
+        // read city name
+
+        std::cout << "Please enter selected city name (with line break after it):" << std::endl;
+
+        auto it = cityToLocation.end();
+
+        while (it == cityToLocation.end())
+        {
+            std::getline(std::cin, city);
+            it = cityToLocation.find(city);
+            if (city == "0")
+            {
+                std::cout << "Bye";
+                return 0;
+            }
+            if (it == cityToLocation.end())
+                std::cout << "ERROR: \"" << city << "\" isn't found in the city list. Please try again." << std::endl;
+            else
+                targetCoordinate = it->second;
+        }
+
+
+        // read radius
+        std::cout << "Please enter the wanted radius :" << std::endl;
+        std::cin >> radius;
+
+
+        // read distance choice calculation
+        std::cout << "Please enter the wanted norm (0 - L2, Euclidean distance, 1 - Linf, Chebyshev distance, 2 - L1, Manhattan distance):" << std::endl;
+        std::cin >> choice;
+
+
+        std::map<std::string, Coordinate> citiesWithinRadius;
+
+
+        // checking for cities within the radius
+        std::copy_if(cityToLocation.begin(), cityToLocation.end(), std::inserter(citiesWithinRadius, citiesWithinRadius.end()),
```

Comment:
you are missing 1 step in finding the cities in radius, you should find vertical and horizontal rectangles, and then to find the intersection square, and to find a circle inside the square.